pkg/distro/defs/README.md

```diff
@@ -0,0 +1,117 @@
+# Image types definitions in YAML
+
+This directory contains the "image-type" definitions in YAML.
+
+We currently have subdirectories for:
+- fedora
+- rhel-7, rhel-8, rhel-9, rhel-10 (which also provide centos,alma)
+
+
+Under each of those directories there is a `distro.yaml` file
+that contains all image-types for the given distro. 
+
+The image types are defined under `image_types` (and for rhel
+also in go-code but that will be generalized soon, it already
+is already for fedora).
+
+## Example distro.yaml
+
+```yaml
+image_types:
+  qcow2: &qcow2
+    image_config: &qcow2_image_config
+      default_target: "multi-user.target"
+      kernel_options: ["console=tty0", "console=ttyS0,115200n8", "no_timer_check"]
+      conditions:
+        "tweak the rhsm config on rhel":
+          when:
+            distro_name: "rhel"
+          shallow_merge:
+            rhsm_config:
+              "no-subscription":
+                dnf_plugin:
+                  product_id:
+                    enabled: false
+                  subscription_manager:
+                    enabled: false
+    partition_table:
+      <<: *default_partition_tables
+    package_sets:
+      os:
+        - &qcow2_pkgset
+          include:
+            - "@core"
+            - ...
+          exclude:
+            - "aic94xx-firmware"
+            - ...
+          conditions:
+            "add insights pkgs on rhel":
+              when:
+                distro_name: "rhel"
+              append:
+                include:
+                  - "insights-client"
+                  - "subscription-manager-cockpit"
+```
+
+Common keys:
+
+### image_config
+
+This maps directly to https://github.com/osbuild/images/blob/v0.154.0/pkg/distro/image_config.go#L18
+
+Conditions can be used and "shallow_merge" operation is used here, this
+means that the image_config from the condition will be merged with
+the original config (but only as a shallow merge).
+
+### partition_table
+
+This maps directly to https://github.com/osbuild/images/blob/v0.154.0/pkg/disk/partition_table.go#L17
+
+Conditions can be used and "override" operation is used here, this
+means that the original partition_table is fully replaced with
+the one found via the condition.
+
+### package_sets
+
+The package sets describe what packages should be included in the
+"os" or "installer" pipelines. Under each keys there is a list of
+objects with "include/exclude" sublists (see the example below).
+
+Conditions can be used and the "append"  operation is used here,
+this means that the packages from the conditions is appended to
+the original package sets.
+
+### conditions
+
+Conditions are expressed using the following form:
+```yaml
+conditions:
+  <<: *shared_conditions
+  "some unique description":
+    when:
+      distro_name: "rhel"
+      arch: "x86_64"
+      version_less_than: "9.2"
+      version_equal: "9.1"
+      version_greater_or_equal: "9.3"
+    shallow_merge|override|append:
```

Comment:
I think this is where people may get hung up (I know I did at first) -- it isn't totally clear that the condition 'action' is only supported by specific keys. I'd do the following:
* In each key section change the wording to say that only the specific action is supported
* Down below the `when` section add a section that says which keys are supported by which actions.